### PR TITLE
all: update kubespray to v2.16.0

### LINKED ISSUE #84: [2] Check possibility to migrate from kubespray 2.15.0 -> 2.16.0
**What should be investigated.**

If we can migrate from kubespray 2.15.0 -> 2.16.0 and if there are any manual steps that are needed (like changing name of some variables or so)

Notable change is that the group `kube-masters` are now called `kube_control_planel`

**What artifacts should this produce.**

Document that describes the upgrade process

### PR DESCRIPTION
**Pull request by viktor-f**
**What this PR does / why we need it**: This updates kubespray to 2.16.0. I have tesed the migration, but I have not tried to init and deploy a new cluster directly at 2.16.0, let me know if you want med to test that as well.

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #84 

**Public facing documentation PR** *(if applicable)*

https://github.com/elastisys/compliantkubernetes/pull/139

**Special notes for reviewer**:

**Checklist:**

- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
  - [ ] is completely transparent, will not impact the workload in any way.
  - [x] requires running a migration script.
  - [ ] requires draining and/or replacing nodes.
  - [ ] will break the cluster.
        I.e. full cluster migration is required.

### CODE REVIEW

File: migration/v2.15.0-ck8s1-v2.16.0-ck8s1/upgrade-cluster.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Upgrade v2.15.0-ck8s1 to v2.16.0-ck8s1
+
+1. Checkout the new release: `git checkout v2.16.0-ck8s1`
+
+1. Update the kubespray submodule: `git submodule update --init --recursive`
+
+1. Kubespray has in 2.16.0 updated the name of their groups.
+    You must manually update the following:
+
+    - If you are not using dynamic inventories (e.g. för Exoscale), change the following groups in your inventory file (`inventory.ini`): kube-master to kube_control_plane, kube-node to kube_node, and k8s-cluster to k8s_cluster.
```
Comment (Xartos) [inline migration/v2.15.0-ck8s1-v2.16.0-ck8s1/upgrade-cluster.md:10]:
```suggestion
    - If you are not using dynamic inventories (e.g. for Exoscale), change the following groups in your inventory file (`inventory.ini`): kube-master to kube_control_plane, kube-node to kube_node, and k8s-cluster to k8s_cluster.
```

**Review by Xartos — APPROVED** [review]
:tada:

**Review by OlleLarsson — APPROVED** [review]

**Comment by viktor-f** [comment]
Updates to public docs https://github.com/elastisys/compliantkubernetes/pull/139

**Review by cristiklein — APPROVED** [review]
Nice!

**Pull request merged**